### PR TITLE
Finish sessions

### LINKED ISSUE #83: Perfect Sessions in CanJS / Feathers / Bitcentive
## Goal

In a template

``` html
<a ($click)="loginGithub()">Github</a>
```

In the view model:

``` js
loginGithub: function(){
    feathersThing.github.loginPopup(whoWeAre);
}
```

In the view model for sessions:

``` js
Session: {value: function(){ return Session }},
session: {
  value: function(resolve){
      return Session.get({}).then(resolve);
  },
  stream: ["Session created", "Session destroyed",
                function(createdStream, destroyedStream, initialValueStream){
      return initialValueStream.merge(createdStream).merge(destroyedStream.map(null))
  }]
}
```

In the Session model:

``` js
sessionConnection = canMap( constructor( feathersSession() ) )

feathers.on("session created", function(session){
    sessionConnection.created(session)
})
feathers.on("session delete", function(session){
    sessionConnection.destroyed(session)
})
```

In the `ContributionMonth` model:

```
canMap( constructor( feathers() ) )

or

superMap([connect.replace(dataUrl, feathers)], {})
```
## Tasks
- [x] - can-event-streams
- [x] - feathers - utilities for starting authentication popup `feathersThing.github.loginPopup`
- [x] - feathers - `featherSession`
  - [x] - SSR document.cookie not populating before app init: https://github.com/donejs/done-ssr/issues/188
  - [x] - Allow feathers-authentication to check cookies after app viewModel loads: https://github.com/feathersjs/feathers-authentication/pull/317
- [ ] Fix this bug with DoneSSR cookies: https://github.com/icanjs/auth-component/issues/15
- [x] - feathers - `feathers` for socket 
  - [ ] - fallback to URL
  - [ ] - How to deal with can-zone XHR AJAX cache? - Make can-zone socket aware.
    - [ ] - steal-socket.io is working in the server when it shouldn't be: https://github.com/stealjs/steal-socket.io/issues/5
- [ ] - async values in `can-define`
- [x] - allow fixture.store like abilities with socket.io. - https://github.com/canjs/can-fixture/issues/55
- [ ] - ecosystem and documented

### PR DESCRIPTION
**Pull request by marshallswain**
Ready for review.

Closes #83, #86, and #87

### CODE REVIEW

**Comment by marshallswain** [comment]
I think https://github.com/donejs/bitcentive/issues/200 is all that's left to wrap this up.

File: public/app.js [hunk 1]
```diff
@@ -22,56 +28,46 @@ const AppViewModel = DefineMap.extend({
...
-      // console.log('streaming');
-      // return canStream.toStream(Session, 'created').onValue(event => {
-      //   let session = event.args[0];
-      //   console.log(session);
-      //   setTimeout(() => {
-      //     self.session = session;
-      //   }, 1);
-      // });
-    // }
+      return canStream.toStream(Session, 'created')
```
Comment (marshallswain) [inline public/app.js]: @justinbmeyer, in this case, using streams was more concise.

File: public/app.js [hunk 2]
```diff
@@ -21,57 +27,59 @@ const AppViewModel = DefineMap.extend({
...
+   * Use Session.get() to see if there's a valid JWT. If one exists, 
+   * a new Session will be created.
    */
-  page: {
-    serialize: true,
-    get(page, setPage){
-      return this.routePage(page);
+  session: {
+    // value () {},
+    stream (setStream) {
```
Comment (justinbmeyer) [inline public/app.js]: remove `setStream`

File: public/app.js [hunk 3]
```diff
@@ -81,54 +89,15 @@ const AppViewModel = DefineMap.extend({
...
-      // Perform some custom logic for logged-in users.
-
-    } else {
-      // Logic for non-authenticated users.
-      if (pageConfig[page] !== 'public') {
-        page = 'login';
+  logout () {
+    this.session && this.session.destroy().then(() => {
+      if (!window.doneSsr) {
+        this.page = 'home';
```
Comment (justinbmeyer) [inline public/app.js]: make it so this logic is incorporated into `page.get`

**Pull request merged**